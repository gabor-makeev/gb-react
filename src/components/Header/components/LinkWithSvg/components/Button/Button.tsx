import styled from 'styled-components';
import { Link } from 'components/Header/components/LinkWithSvg/components/Link/Link';

export const Button = styled(Link)`
  font: inherit;
  letter-spacing: inherit;
  border: none;
  cursor: pointer;
`;
