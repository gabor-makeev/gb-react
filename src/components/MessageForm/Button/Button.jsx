import style from './Button.module.scss';
import MaterialButton from '@mui/material/Button';

export const Button = ({ disabled }) => {
  return (
    <MaterialButton
      className={style.button}
      disabled={disabled}
      type={'submit'}
      data-testid={'message-send-button'}
    >
      Send
    </MaterialButton>
  );
};
