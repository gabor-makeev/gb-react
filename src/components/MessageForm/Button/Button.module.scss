.button {
  max-width: 55px;
  align-self: center;
  padding: 10px;
  border: none;
  border-radius: 15px;
  background-color: #31E865;
  transition: 0.3s;

  &:hover {
    box-shadow: rgba(0, 0, 0, 0.35) 0 5px 15px;
  }

  &:disabled:hover {
    box-shadow: none;
  }
}