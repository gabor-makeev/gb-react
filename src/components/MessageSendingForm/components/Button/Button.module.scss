.button {
  //Custom styles should go here
}