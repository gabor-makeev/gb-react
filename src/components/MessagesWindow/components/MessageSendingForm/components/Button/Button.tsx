import { FC } from 'react';
import style from './Button.module.scss';
import MaterialButton from '@mui/material/Button';

interface ButtonProps {
  disabled?: boolean;
}

export const Button: FC<ButtonProps> = ({ disabled = false }) => {
  return (
    <MaterialButton
      className={style.button}
      disabled={disabled}
      type={'submit'}
      data-testid={'messageSendingFormButton'}
    >
      Send
    </MaterialButton>
  );
};
