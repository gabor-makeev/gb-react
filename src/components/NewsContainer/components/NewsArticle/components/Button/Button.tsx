import styled from 'styled-components';

export const Button = styled.button`
  font-size: 18px;
  font-weight: 600;
  line-height: 24px;
  letter-spacing: -0.5px;

  @media (max-width: 1024px) {
    font-size: 16px;
  }
  @media (max-width: 768px) {
    font-size: 14px;
  }
`;
