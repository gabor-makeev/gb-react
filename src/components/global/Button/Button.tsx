import styled from 'styled-components';

export const Button = styled.button`
  max-width: 343px;
  width: 100%;
  padding: 16px 8px;
  transition: 0.3s;
  cursor: pointer;
  border: none;
  border-radius: 16px;
  background-color: #244c66;
  font-family: 'Inter', sans-serif;
  font-weight: 700;
  font-size: 16px;
  line-height: 24px;
  letter-spacing: -0.011em;
  color: white;

  &:hover {
    background-color: transparent;
    color: #244c66;
    box-shadow: inset 0 0 0 1.5px #244c66;
  }

  &:active {
    background-color: #244c66;
    color: white;
  }
`;
