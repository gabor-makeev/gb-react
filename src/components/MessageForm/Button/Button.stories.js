import { Button } from './Button';

export default {
  title: 'Button',
  component: Button,
};

const Template = (args) => <Button {...args} />;

export const Primary = Template.bind({});

export const Disabled = Template.bind({});
Disabled.args = {
  disabled: true,
};
