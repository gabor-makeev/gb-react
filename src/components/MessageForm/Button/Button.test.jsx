import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { userEvent } from '@storybook/testing-library';

import { Button } from './Button';

describe('Button', () => {
  it('should render', () => {
    render(<Button />);
  });

  it('should be disabled', () => {
    render(<Button disabled={true} />);

    expect(screen.getByRole('button')).toBeDisabled();
  });

  it('should render a button element', () => {
    render(<Button />);

    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  it('should get clicked', () => {
    const mockHandler = jest.fn();

    render(<Button />);
    const button = screen.getByRole('button');
    button.onClick = mockHandler();

    userEvent.click(button);
    expect(mockHandler).toBeCalledTimes(1);
  });

  it('should render with snapshot', () => {
    const { asFragment } = render(<Button />);

    expect(asFragment()).toMatchSnapshot();
  });
});
