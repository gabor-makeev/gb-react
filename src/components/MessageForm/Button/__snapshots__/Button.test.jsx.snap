// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
  >
    Send
  </button>
</DocumentFragment>
`;
